doc/source/cluster/kubernetes/user-guides/logging.md

```diff
@@ -242,6 +244,118 @@ Learn how to write filters in this [filter pattern syntax doc](https://docs.aws.
+* Outputs: The `loki` output block specifies Loki as the target. The `Host` and `Port` define the Loki service endpoint, and `Labels` add metadata for easier querying in Grafana. Additionally, `tenant_id` allows for multi-tenancy if required by the Loki setup.
+
+Deploy the Fluent Bit deployment with the [Helm chart repository](https://github.com/fluent/helm-charts/tree/main/charts/fluent-bit).
+
+```shell
+helm repo add fluent https://fluent.github.io/helm-charts
+helm repo update
+
+helm install fluent-bit fluent/fluent-bit -f fluent-bit-config.yaml
+```
+
+#### Deploy a KubeRay operator and RayCluster
+
+Deploy the KubeRay operator and RayCluster with the [Helm chart repository](https://github.com/ray-project/kuberay-helm).
+
+```shell
+helm repo add kuberay https://ray-project.github.io/kuberay-helm/
+helm repo update
+
+helm install kuberay-operator kuberay/kuberay-operator --version 1.2.2
+
+# Confirm that the operator is running in the namespace `default`.
+kubectl get pods
+# NAME                                READY   STATUS    RESTARTS   AGE
+# kuberay-operator-7fbdbf8c89-pt8bk   1/1     Running   0          27s
+
+helm install raycluster kuberay/ray-cluster --version 1.2.2
+
+# Once the RayCluster CR has been created, you can view it by running:
+kubectl get rayclusters
+
+# NAME                 DESIRED WORKERS   AVAILABLE WORKERS   CPUS   MEMORY   GPUS   STATUS   AGE
+# raycluster-kuberay   1                 1                   2      3G       0      ready    95s
+```
+
+#### Deploy Grafana
+
+Create a `datasource-config.yaml` file with the following configuration to set up Grafana's Loki datasource:
+```{literalinclude} ../configs/loki.log.yaml
+:language: yaml
+:start-after: Grafana Datasource Config
+```
+
+Deploy the Grafana deployment with the [Helm chart repository](https://github.com/grafana/helm-charts/tree/main/charts/grafana).
+
+```shell
+helm repo add grafana https://grafana.github.io/helm-charts
+helm repo update
+
+helm install grafana grafana/grafana -f datasource-config.yaml
+```
+
+#### Check the Grafana Dashboard
+
+Get the admin user password by running the following command:
+
+```shell
+kubectl get secret --namespace default grafana -o jsonpath="{.data.admin-password}" | base64 --decode ; echo
+```
+
+To access Grafana from your local machine, set up port forwarding by running:
+```shell
+export POD_NAME=$(kubectl get pods --namespace default -l "app.kubernetes.io/name=grafana,app.kubernetes.io/instance=grafana" -o jsonpath="{.items[0].metadata.name}")
+kubectl --namespace default port-forward $POD_NAME 3000
+```
+
+This will make Grafana available locally at `http://localhost:3000`.
+
+Finally, use the LogQL query to view logs for a specific pod, such as the kuberay-operator:
+
+```
+{pod="kuberay-operator-xxxxxxxx-xxxxx"}
+```
+
+![Loki Logs](images/loki-logs.png)
```

Comment:
In the screenshot, it looks like string matching is being applied to the entire log message. Is it possible to restrict the string matching to `RayCluster.name` only?

Reply:
Sure, I can add your suggested method to the Persist KubeRay Operator Logs documentation, as this approach will preserve all pod logs, including those from RayCluster, RayService, and others.

Reply:
This method currently retains all pod stdout/stderr logs. If Fluent Bit adds the RayCluster.name label when sending logs to Loki for filtering, it may result in the inability to search for logs from other unrelated pods. I can experiment with this to see the impact.

Reply:
I will try to see if it's possible to add this custom label.